src/go/k8s/pkg/resources/statefulset_scale.go

```diff
@@ -140,22 +111,66 @@ func (r *StatefulSetResource) handleScaling(ctx context.Context) error {
 	}
 
 	// User required replicas is lower than current replicas (currentReplicas): start the decommissioning process
+	r.logger.Info("Downscaling cluster", "replicas", *r.pandaCluster.Spec.Replicas)
+
 	targetOrdinal := r.pandaCluster.Status.CurrentReplicas - 1 // Always decommission last node
 	targetBroker, err := r.getBrokerIDForPod(ctx, targetOrdinal)
 	if err != nil {
-		return err
+		return fmt.Errorf("error getting broker ID for pod %v when downscaling cluster: %w", targetOrdinal, err)
 	}
 	if targetBroker == nil {
-		return &RequeueAfterError{
-			RequeueAfter: RequeueDuration,
-			Msg:          fmt.Sprintf("cannot retrieve broker id for pod %d", targetOrdinal),
-		}
+		// The target pod isn't in the broker list. Just select a non-existing broker for decommission so the next
+		// reconcile loop will succeed.
+		nonExistantBroker := int32(-1)
+		targetBroker = &nonExistantBroker
```

Comment:
This was left-over from when the ids were the pod ordinals.

Reply:
The requeue after error play key role to speed up next reconciliation. When you move with targetBroker set to `-1` the update of the Cluster custom resource status field would not error out (I think). The next reconciliation will be around 1h as far as I remember. 

Another question is why not `nil` like in
https://github.com/redpanda-data/redpanda/blame/41778312057e4bb40c9465ad79e3689995bfcd2d/src/go/k8s/pkg/resources/statefulset_scale.go#L187-L191

Reply:
> requeue after error

good point.

> why not nil? 

1. because I didn't know what =nil meant. nil usually means unset so there's nothing to decommission and, in fact, everything that checks to see if it's nil returns immediately without doing anything. 
2. Since we want to reach the code where the pod is deleted, it's important that we can think we're decommissioning some broker, and that we can discover that the broker doesn't exist so we can delete the pod.

Reply:
added a fix for the first point